src/main/scala/com/snowflake/snowpark/internal/ParameterUtils.scala

```diff
@@ -106,36 +106,45 @@ private[snowpark] object ParameterUtils extends Logging {
   private[snowpark] def parsePrivateKey(key: String): PrivateKey = {
     try {
       val decoded = Base64.decodeBase64(key)
-      val derReader = new DerInputStream(decoded)
-      val seq = derReader.getSequence(0)
-
-      if (seq.length < 9) {
-        throw new GeneralSecurityException("Could not parse a PKCS1 private key.")
-      }
-
-      // seq(0) is version, skip
-      val modulus = seq(1).getBigInteger
-      val publicExp = seq(2).getBigInteger
-      val privateExp = seq(3).getBigInteger
-      val prime1 = seq(4).getBigInteger
-      val prime2 = seq(5).getBigInteger
-      val exp1 = seq(6).getBigInteger
-      val exp2 = seq(7).getBigInteger
-      val crtCoef = seq(8).getBigInteger
-      val keySpec = new RSAPrivateCrtKeySpec(
-        modulus,
-        publicExp,
-        privateExp,
-        prime1,
-        prime2,
-        exp1,
-        exp2,
-        crtCoef)
-      val keyFactory = KeyFactory.getInstance("RSA")
-      keyFactory.generatePrivate(keySpec)
+      val kf = KeyFactory.getInstance("RSA")
```

Comment:
NIT: Can you add a comment for the function to explain that:
It first attempts to decode as RSA, if failed, fall back to decode as PKSC1

Reply:
added